File: assignmentOne.pl
parent('siddique','masum').
parent('siddique','laila').
parent('masum','raiyan').
parent('raiyan','rafsan').
parent('raiyan','ebon').
parent('raiyan','raisa').
parent('rafsan','shihab').

male('siddique'). male('masum'). male('raiyan'). male('rafsan').
male('ebon'). male('shihab'). female('laila'). female('raisa').

brother(X,Y):- parent(Z,X), parent(Z,Y), not(X=Y), male(X).
%We assume X is brother of Y


printbrother :- write('Person: '), read(BR),
	  write('Brother: '), brother(B, BR),
	  write(B), tab(5), fail.
printbrother.

sister(X,Y):- parent(Z,X), parent(Z,Y), not(X=Y), female(X).
%We assume X is sister of Y

printsister :- write('Person: '), read(SS),
	  write('Sister: '), sister(S, SS),
	  write(S), tab(5), fail.
printsister.

uncle(X,Y):- parent(Z,Y), brother(X,Z), male(X).
%We assume X is uncle, Y is nephew/niece

printuncle :- write('Person: '), read(UNC),
	  write('Uncle: '), uncle(UN, UNC),
	  write(UN), tab(5), fail.
printuncle.

aunt(X,Y):- parent(Z,Y), sister(X,Z), female(X).
%We assume X is aunt, Y is nephew/niece

printaunt:- write('Person: '), read(AUN),
	  write('Aunt: '), aunt(AU, AUN),
	  write(AU), tab(5), fail.
printaunt.

File: assignmentOne.py
male = ['siddique','masum','raiyan','rafsan','ebon','shihab']
female = ['laila','raisa']
parent = [
('siddique','masum'),('siddique','laila'),('masum','raiyan'),('raiyan','rafsan'),
('raiyan','ebon'),('raiyan','raisa'),('rafsan','shihab')
]

#Grandparent 

def printgrandparent():
    Y = input('Person:  ')
    print('Grandparent:', end='  ')
    n = len(parent)
    for i in range(n):
        if parent[i][1] == Y:
            for j in range(n):
                if parent[i][0] == parent[j][1]:
                    print(parent[j][0], end='  ')
        i+=1

#Brother 

def printbrother():
    Y = input('Person:  ')
    print('Brother:', end='  ')
    n = len(parent)
    for i in range(n):
        if parent[i][1] == Y:
            for j in range(n):
                if (parent[i][0] == parent[j][0]) and (
                        parent[i][1] != parent[j][1]) and (parent[j][1] in male):
                   print(parent[j][1], end='  ')
        i+=1                   

#Sister 

def printsister():
    Y = input('Person:  ')
    print('Sister:', end='  ')
    n = len(parent)
    for i in range(n):
        if parent[i][1] == Y:
            for j in range(n):
                if (parent[i][0] == parent[j][0]) and (
                        parent[i][1] != parent[j][1]) and (parent[j][1] in female):
                   print(parent[j][1], end='  ')
        i+=1

#Uncle 
def printuncle():
    Y = input('Person:  ')
    print('Uncle:', end='  ')
    n = len(parent)
    for i in range(n):
        if parent[i][1] == Y:
            for j in range(n):
                if parent[i][0] == parent[j][1]:
                    for k in range(n):
                        if (parent[k][0] == parent[j][0]) and ( 
                                parent[k][1] != parent[j][1]) and (parent[k][1] in male):
                            print(parent[k][1], end='  ')
        i+=1
        

#Aunt 

def printaunt():
    Y = input('Person:  ')
    print('Aunt:', end='  ')
    n = len(parent)
    for i in range(n):
        if parent[i][1] == Y:
            for j in range(n):
                if parent[i][0] == parent[j][1]:
                    for k in range(n):
                        if (parent[k][0] == parent[j][0]) and ( 
                                parent[k][1] != parent[j][1]) and (parent[k][1] in female):
                            print(parent[k][1], end='  ')
        i+=1


printgrandparent()
printbrother()
printsister()
printuncle()
printaunt()
